### PR TITLE
Bug 1254597 browser toolbar menu

### PR DESCRIPTION
**Pull request by fluffyemily**
Update the browser toolbar to show the menu not the bookmarks
build behind the menu flag so we can still use bookmark flag if we don't want to ship menu

https://bugzilla.mozilla.org/show_bug.cgi?id=1254597

### CODE REVIEW

File: Client/Frontend/Browser/BrowserToolbar.swift [hunk 1]
```diff
@@ -78,42 +80,49 @@ public class BrowserToolbarHelper: NSObject {
...
-        toolbar.bookmarkButton.setImage(UIImage(named: "bookmarkHighlighted"), forState: UIControlState.Highlighted)
-        toolbar.bookmarkButton.accessibilityLabel = NSLocalizedString("Bookmark", comment: "Accessibility Label for the browser toolbar Bookmark button")
-        let longPressGestureBookmarkButton = UILongPressGestureRecognizer(target: self, action: #selector(BrowserToolbarHelper.SELdidLongPressBookmark(_:)))
-        toolbar.bookmarkButton.addGestureRecognizer(longPressGestureBookmarkButton)
-        toolbar.bookmarkButton.addTarget(self, action: #selector(BrowserToolbarHelper.SELdidClickBookmark), forControlEvents: UIControlEvents.TouchUpInside)
+        if AppConstants.MOZ_MENU {
+            toolbar.menuButton.contentMode = UIViewContentMode.Center
+            toolbar.menuButton.setImage(UIImage.templateImageNamed("bottomNav-menu"), forState: .Normal)
+            toolbar.menuButton.accessibilityLabel = NSLocalizedString("Menu", comment: "Accessibility Label for the browser toolbar Menu button")
+            toolbar.menuButton.addTarget(self, action: #selector(BrowserToolbarHelper.SELdidClickMenu), forControlEvents: UIControlEvents.TouchUpInside)
```
Comment (sleroux) [inline Client/Frontend/Browser/BrowserToolbar.swift]: Nit: With the new #selector form, would it make sense to start dropping the SEL prefix?

Comment (sleroux) [inline Client/Frontend/Browser/BrowserToolbar.swift]: Nevermind - makes sense to follow the pattern here and change it later.

**Comment by sleroux** [comment]
Code looks good! Found a couple of bugs though (probably already aware of them from the menu work):

1. When opening the menu in landscape iPhone and transitioning back to portrait, the popover appears when it shouldn't. Not sure if we want to dismiss the popover and display the modal instead in this case or dismiss the menu.

2. When displaying the menu, the shadow background animates up along with the modal. Probably something WIP in the menu branch so you can probably ignore this comment.

Long term, I think we might want to replace the BrowserToolbar we currently have with he native `UIToolbar` widget. I tried using it for the Tab Tray (https://github.com/mozilla/firefox-ios/pull/1670) and I found it simplified the code quite a bit since it handles the layout for us. I'd love to get your opinion on it and pros/cons.

**Comment by fluffyemily** [comment]
@sleroux as far as replacing the BrowserToolbar with something based on UIToolbar, I totally agree it's the way to go. The tricky thing will be doing it in such a way that, when in landscape/ipad we can still create a toolbar that can contain a URLBarView. I feel we can - it would have to be a UIBarButtonItem but I've stuck UITextField's in UIToolbars that way before and it's always worked out OK.

Stage 1, get all this working with the existing toolbars. Stage 2, rewrite toolbars to make them saner!

**Comment by jhugman** [comment]
Two WIP things, more tecgirl's remit than mine: 

1. sleroux has already highlighted: the shade coming in from the bottom. Would suggest a fade in.
2. Pressing the menu button when the menu is open should… dismiss the menu?

**Comment by fluffyemily** [comment]
@jhugman am sorting the transition. And yes, pressing the menu button should dismiss the menu. That used to work. Will investigate why it is no longer working

**Comment by fluffyemily** [comment]
@jhugman &@sleroux Updates made as per comments/bugs found.

re: the menu shade. I'm going to totally revise the whole presentation animation thing at some point, so I'm not going to play around with that too much now. I'll almost certainly require some @sleroux animation guidance there.

**Comment by fluffyemily** [comment]
closed in favour of https://github.com/mozilla/firefox-ios/pull/1675

**Pull request closed without merging**